spec/rails_helper.rb

```diff
@@ -124,11 +124,33 @@
 
   end
 
+  config.after(:each) do |ex|
+    # I hope this would be easy to maintain in the long run, but it is optional.
+    # Just to remind anybody trying to use Mocha expectations in specs that this is not mandated.
+    has_expectations = Mocha::Mockery.instance.mocks.any? do |mock|
+      mock.__expectations__.to_a.any? do |expectation|
+        !expectation.instance_variable_get(:@cardinality).allowed_any_number_of_times?
+      end
+    end
```

Comment:
This seems to access Mocha internal variables and methods. If they change any of these in a next release, this will stop working. We'll have to be aware of this every time we want to update mocha, which I think we'll forget. I would look for another way to implement this check, or remove it if it's optional.

Reply:
It is optional. If api changes, then test suite would fail. And we will know. So my line of thinking is that we can keep it until it becomes a problem and remove if we want later. Maybe comment not clear enough?